homeassistant/components/wled/__init__.py

```diff
@@ -0,0 +1,175 @@
+"""Support for WLED."""
+from datetime import timedelta
+import logging
+from typing import Any, Dict, Optional, Union
+
+from wled import WLED, WLEDConnectionError, WLEDError
+
+from homeassistant.components.light import DOMAIN as LIGHT_DOMAIN
+from homeassistant.components.wled.const import (
+    ATTR_IDENTIFIERS,
+    ATTR_MANUFACTURER,
+    ATTR_MODEL,
+    ATTR_SOFTWARE_VERSION,
+    DATA_WLED_CLIENT,
+    DATA_WLED_UPDATED,
+    DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_NAME, CONF_HOST
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.dispatcher import async_dispatcher_connect, dispatcher_send
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.util import dt as dt_util
+
+SCAN_INTERVAL = timedelta(seconds=5)
+PARALLEL_UPDATES = 1
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the WLED components."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up WLED from a config entry."""
+
+    # Create WLED instance for this entry
+    session = async_get_clientsession(hass)
+    wled = WLED(entry.data[CONF_HOST], loop=hass.loop, session=session)
+
+    # Ensure we can connect and talk to it
+    try:
+        await wled.update()
+    except WLEDConnectionError as exception:
+        raise ConfigEntryNotReady from exception
+
+    hass.data.setdefault(DOMAIN, {}).setdefault(entry.entry_id, {})
```

Comment:
Do we need to use `setdefault` for the entry.entry_id? Will we ever set up the entry twice while the entry id is still in hass.data? That looks like a bug in that case.